src/unix/battery.c

```diff
@@ -0,0 +1,252 @@
+/* Copyright libuv project contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+
+#if defined(__APPLE__) || TARGET_OS_IPHONE
+#include <dlfcn.h>
+#include "darwin-stub.h"
+#endif // defined(__APPLE__) || TARGET_OS_IPHONE
+
+// macOS won't need any of this
+#if defined(__LINUX__)
+#include <stdlib.h>
+#include "internal.h"
+#endif
+
+int uv_battery_info(uv_battery_info_t* info) {
+  int err;
+  err = UV_ENOSYS;
+#if defined(__APPLE__) || TARGET_OS_IPHONE
+  CFTypeRef (*pIOPSCopyPowerSourcesInfo)(void);
+  CFArrayRef (*pIOPSCopyPowerSourcesList)(CFTypeRef);
+  CFIndex (*pCFArrayGetCount)(CFArrayRef);
+  CFArrayRef (*pIOPSGetPowerSourceDescription)(CFTypeRef, CFTypeRef);
+  void (*pCFRelease)(CFTypeRef);
+  void (*pCFNumberGetValue)(CFNumberRef, int, void *);
+  void* (*pCFArrayGetValueAtIndex)(CFArrayRef, CFIndex);
+  void* (*pCFDictionaryGetValue)(CFDictionaryRef, const void*);
+  int (*pCFBooleanGetValue)(CFBooleanRef);
+  CFStringRef (*pCFStringCreateWithCString)(CFAllocatorRef,
+                                            const char*,
+                                            CFStringEncoding);
+
+  CFBooleanRef isChargingValue;
+  CFDictionaryRef powerSource;
+  CFNumberRef currentCapacity;
+  CFNumberRef timeRemainingValue;
```

Comment:
Libuv uses snake_case for locals, ex. `is_charging_value`, `power_source`, etc.

(`pCFArrayGetCount` and friends get a pass since they mirror Apple's naming convention.)